### PR TITLE
[fake autoscaler] use check_call in fake multi node test utils

### PR DESCRIPTION
**Pull request by aslonnie**
so that output is printed to logs

and also use "sys.executable" rather than "python"

### CODE REVIEW

**Review by hongpeng-guo — APPROVED** [review]
Thanks for improve the observability of this test!

Nits: I noticed two more places using `subprocess.check_output` without getting its return value in the `fake_multi_node` directory. I guess it makes sense to replace them as `subprocess.check_call` as well. These are:
1. https://github.com/anyscale/runtime/blob/626130fe7a2e32775dcf092eef6b60ee67c1ad19/python/ray/autoscaler/_private/fake_multi_node/docker_monitor.py#L68-L74
2. https://github.com/anyscale/runtime/blob/626130fe7a2e32775dcf092eef6b60ee67c1ad19/python/ray/autoscaler/_private/fake_multi_node/node_provider.py#L468-L484

Feel free to change if it makes sense to you, too.

**Comment by aslonnie** [comment]
> Feel free to change if it makes sense to you, too.

changed those too.

File: python/ray/autoscaler/_private/fake_multi_node/docker_monitor.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def _update_docker_status(
     data_str = ""
     try:
         data_str = (
-            subprocess.check_output(
+            subprocess.check_call(
```
Comment (hongpeng-guo) [inline python/ray/autoscaler/_private/fake_multi_node/docker_monitor.py:113]: Probably not this one. This call requires `data_str` return value from `check_output`

Comment (aslonnie) [reply python/ray/autoscaler/_private/fake_multi_node/docker_monitor.py:113]: true..

**Pull request merged**